custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js

```diff
@@ -337,17 +345,123 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
         }
     };
 
+    // mobile helpers
+    vm.showSortPopup = false;
+    vm.tooltipDisplayed = -1; // '-1' when none of the tooltips are displayed in sort popup
+    vm.requestDataStartingFrom = 0; // could be any multiple of 10
+    vm.dataSortingDirection = 0; // '0' -> asc, '1' -> desc
+    vm.sortingColumn = 0;
+    vm.sortableInputKpiData = [];
+    vm.showSortingInfo = function (tooltipNumber) {
+        // when clicked on 'i' icon, get the column number on which info is requested and show only corresponding info.
+        vm.tooltipDisplayed = tooltipNumber;
+    };
+    vm.toggleSortPopup = function (event) {
+        vm.tooltipDisplayed = -1;
+        vm.showSortPopup = !vm.showSortPopup;
+        event.stopPropagation();
+    };
+    vm.clearSorting = function (event) {
+        // resets to sorting by location name and closes sort popup
+        vm.dataSortingDirection = 0;
+        vm.sortingColumn = 0;
+        vm.sortableInputKpiData = [];
+        vm.requestDataStartingFrom = 0;
+        vm.getData();
+        vm.toggleSortPopup(event);
+    };
+    vm.getMobileData = function (index) {
+        // triggers when clicked on any of the headings in sort popup
+        vm.dataSortingDirection = 1 - vm.dataSortingDirection;
+        vm.sortingColumn = index + 1;
```

Comment:
not sure i understand this. why does the column and direction change? wouldn't direction only change if the column is the same?

Reply:
Correct. I will update the logic